rlclientlib/live_model_impl.cc

```diff
@@ -208,11 +208,15 @@ namespace reinforcement_learning {
     RETURN_IF_FAIL(check_null_or_empty(event_id, _trace_logger.get(), status));
     RETURN_IF_FAIL(check_null_or_empty(context_json, _trace_logger.get(), status));
 
+    utility::ContextInfo context_info;
+    RETURN_IF_FAIL(utility::get_context_info(context_json, context_info, _trace_logger.get(), status));
+
     // Ensure multi comes before slots, this is a current limitation of the parser.
-    RETURN_IF_FAIL(utility::validate_multi_before_slots(context_json, _trace_logger.get(), status));
+    if(context_info.slots.size() < 1 || context_info.actions.size() < 1 || context_info.slots[0].first < context_info.actions[0].first) {
+      RETURN_ERROR_LS(_trace_logger.get(), status, json_parse_error) << "There must be both a _multi field and _slots, and _multi must come first.";
+    }
 
-    size_t num_decisions;
-    RETURN_IF_FAIL(utility::get_slot_count(num_decisions, context_json, _trace_logger.get(), status));
```

Comment:
same question as about validate_multi_before_slots

Reply:
All those functions would go away as all we need from them is provided by ContextInfo.
I'd probably enrich ContextInfo to include relevant helpers such as `is_multi_before_slots`.
One thing I need to add is id extraction for slots/actions for it to completely subsumes all the helpers we have.